### PR TITLE
Add typed sub-feature settings

### PR DESCRIPTION
**Pull request by sammacbeth**
**Asana Task/Github Issue:** https://app.asana.com/0/1206780406371845/1208746808255102/f

## Description
- Add a `settings` property to sub-features ("feature flags").
- This property will default to `Record<string, string>`, but can be optionally explicitly typed.
- Added an example of explicit typing for Autofill, and the `canImportFromGooglePasswordManager` sub-feature, from https://github.com/duckduckgo/privacy-configuration/pull/2453.

#### Reference
- [Config Reviewer Documentation](https://app.asana.com/0/1200890834746050/1204443212791216/f)
- [Config Maintainer Documentation](https://app.asana.com/0/1200890834746050/1200573250322769/f)
- [Feature Implementer Documentation](https://app.asana.com/0/1200890834746050/1201498956177210/f)

### CODE REVIEW

**Review by jonathanKingston — APPROVED** [review]

**Review by jonathanKingston — APPROVED** [review]

File: schema/features/autofill.ts
Comment (jonathanKingston) [inline schema/features/autofill.ts:13]: We still will want to change this once the feature gets used, fine to merge as is though.

Comment (sammacbeth) [reply schema/features/autofill.ts:13]: Yeah, currently waiting for @CDRussell to confirm what this schema should be.

Comment (CDRussell) [reply schema/features/autofill.ts:13]: Feel free to merge this without waiting on me, and if we need to update this again it will be done as part of https://github.com/duckduckgo/privacy-configuration/pull/2453

Comment (jonathanKingston) [reply schema/features/autofill.ts:13]: I think it will be probably:
```js
            javascriptConfig: CSSInjectFeatureSettings<ImportFromGooglePasswordManager>;
```

Where:
```ts
type ButtonConfig = {
  shouldAutotap: boolean;
  path: string;
  selectors: string[];
  labelTexts: string[];
};

type ImportFromGooglePasswordManager = {
  settingsButton: ButtonConfig;
  exportButton: ButtonConfig;
  signInButton: ButtonConfig;
};
```

Assuming the config looks like:
```json
"javascriptConfig": {
  "domains": [
    {
      "domain": [
        "google.com",
        "localhost"
      ],
      "patchSettings": [
        {
          "path": "",
          "op": "add",
          "value": {
            "settingsButton": {
              "shouldAutotap": false,
              "path": "/",
              "selectors": [
                "a[href*='options']"
              ],
              "labelTexts": [
                "Password options"
              ]
            },
            "exportButton": {
              "shouldAutotap": false,
              "path": "/options",
              "selectors": [
                "c-wiz[data-node-index*='2;0'][data-p*='options']",
                "c-wiz[data-p*='options'][jsdata='deferred-i4']"
              ],
              "labelTexts": [
                "Export"
              ]
            },
            "signInButton": {
              "shouldAutotap": false,
              "path": "/intro",
              "selectors": [
                "a[href*='ServiceLogin'][target='_top']"
              ],
              "labelTexts": [
                "Sign in"
              ]
            }
          }
        }
      ]
    }
  ]
}

```

Comment (CDRussell) [reply schema/features/autofill.ts:13]: Looks good to me

File: schema/features/autofill.ts
Comment (jonathanKingston) [inline schema/features/autofill.ts:11]: Rerun prettier format I think is the errors.

**Review by jonathanKingston — APPROVED** [review]

**Pull request merged**
